codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java

```diff
@@ -71,6 +66,9 @@ public MqttDecoder(int maxBytesInMessage) {
 
     @Override
     protected void decode(ChannelHandlerContext ctx, ByteBuf buffer, List<Object> out) throws Exception {
+        MqttFixedHeader mqttFixedHeader = null;
+        Object variableHeader = null;
+        int bytesRemainingInVariablePart = 0;
```

Comment:
it's impossible because variables initialized in one `case` and using in the other..